chains/substrate/listener/listener.go

```diff
@@ -82,16 +87,16 @@ func (l *SubstrateListener) ListenToEvents(ctx context.Context, startBlock *big.
 					time.Sleep(l.blockRetryInterval)
 					continue
 				}
-
-				evts, err := l.fetchEvents(startBlock, endBlock)
+				// This context will share context between fetchEvents and HandleEvents. Might be a better idea to refactor it in future.
```

Comment:
Why is that a problem and how would we refactor it?

Reply:
this comment does not make any sense now, leftovers from my thinking process